### PR TITLE
feat: add stream method for `ServerStream`

### LINKED ISSUE #1576: `ServerStream` do not easily support Java Streams
Add stream support for `com.google.api.gax.rpc.ServerStream`.

### PR DESCRIPTION
**Pull request by JoeWang1127**
Fixes #1576 ☕️

### CODE REVIEW

**Comment by burkedavison** [comment]
LGTM

**Review by suztomo — APPROVED** [review]
Let's merge this at the same time as the Page's

File: gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java [hunk 1]
```diff
@@ -89,6 +91,10 @@ public Iterator<V> iterator() {
     return iterator;
   }
 
+  public Stream<V> stream() {
```
Comment (blakeli0) [inline gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java:94]: Can you please add Javadoc? All public methods are required to have Javadoc.

File: gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java [hunk 2]
```diff
@@ -89,6 +91,10 @@ public Iterator<V> iterator() {
...
+    return StreamSupport.stream(this.spliterator(), false);
```
Comment (blakeli0) [inline gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java:95]: I see that the second argument is set to false, which means this is a sequential stream, do we want to expose another `parallelStream` method as well?

Comment (JoeWang1127) [reply gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java:95]: I'm not sure whether a deterministic result is required as parallel stream can randomize the result.

File: gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java (hunk 1 above)
Comment (JoeWang1127) [reply gax-java/gax/src/main/java/com/google/api/gax/rpc/ServerStream.java:99]: done

**Pull request merged**